### PR TITLE
make sure that we do not use any unnecessary dust 

### PR DESCRIPTION
**Pull request by jack60612**

### CODE REVIEW

**Comment by jack60612** [comment]
test failure is unrelated

File: tests/wallet/test_coin_selection.py [hunk 1]
```diff
@@ -108,7 +108,28 @@ async def test_coin_selection_with_dust(self, a_hash: bytes32) -> None:
...
+                spendable_amount,
+                DEFAULT_CONSTANTS.MAX_COIN_AMOUNT,
+                coin_list,
+                {},
+                logging.getLogger("test"),
+                uint128(target_amount),
+            )
+            assert dusty_result is not None
+            assert sum([coin.amount for coin in dusty_result]) >= target_amount
+            assert target_amount == len(dusty_result) * 2000  # make sure that we do not use any dust coins.
```
Comment (mariano54) [inline tests/wallet/test_coin_selection.py:131]: This doesn't prove that, I think we need to do:

```
for coin in dusty_result:
    assert coin.amount > 1
```

File: tests/wallet/test_coin_selection.py [hunk 2]
```diff
@@ -108,7 +108,28 @@ async def test_coin_selection_with_dust(self, a_hash: bytes32) -> None:
...
+            assert len(result) == 1  # only one coin should be selected
+
+        for i in range(100):
+            coin_list.append(
+                WalletCoinRecord(
+                    Coin(a_hash, std_hash(i), uint64(2000)), uint32(1), uint32(1), False, True, WalletType(0), 1
+                )
+            )
+        spendable_amount = uint128(spendable_amount + 2000 * 100)
+        for target_amount in [50000, 10000]:  # select the first 100 values
```
Comment (mariano54) [inline tests/wallet/test_coin_selection.py:120]: Try some other targets that don't make us choose exact amount as well

File: tests/wallet/test_coin_selection.py (hunk 1 above)
Comment (jack60612) [reply tests/wallet/test_coin_selection.py:131]: Will do, i was trying to avoid that

File: tests/wallet/test_coin_selection.py (hunk 2 above)
Comment (jack60612) [reply tests/wallet/test_coin_selection.py:120]: gotcha.

**Comment by jack60612** [comment]
oops

**Pull request merged**